apps/rush-lib/src/cli/utilities/pnpm/PnpmLinkManager.ts

```diff
@@ -0,0 +1,221 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as fsx from 'fs-extra';
+import * as os from 'os';
+import * as path from 'path';
+import * as semver from 'semver';
+import * as tar from 'tar';
+import encodeRegistry = require('encode-registry');
+
+import { JsonFile } from '@microsoft/node-core-library';
+
+import LinkManager, {
+  SymlinkKind
+} from '../LinkManager';
+import Utilities from '../../../utilities/Utilities';
+import Package from '../Package';
+import { RushConstants } from '../../../RushConstants';
+import { IRushLinkJson } from '../../../data/RushConfiguration';
+import RushConfigurationProject from '../../../data/RushConfigurationProject';
+
+export default class PnpmLinkManager extends LinkManager {
+  protected _linkProjects(): Promise<void> {
+    return new Promise<void>((resolve: () => void, reject: (reason: Error) => void): void => {
+      // go ahead and find out the registry we are using, we need this to locate the pnpm store folder
+      const registryUrl: string = Utilities.executeCommandAndCaptureOutput(
+        this._rushConfiguration.packageManagerToolFilename,
+        ['config', 'get', 'registry'],
+        this._rushConfiguration.commonTempFolder,
+        process.env);
+
+      const encodedRegistry: string = encodeRegistry(registryUrl);
+
+      const registryPath: string = path.join(
```

Comment:
>registryPath [](start = 12, length = 12)

On your whiteboard we started some nomenclature for PNPM.  Its directory layout is very complex.  We need consistent technical names for all the interest parts of it, so that our code and comments are understandable.  Could you put this in the source code somewhere, e.g. as comments or constants or something?